internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -155,12 +150,24 @@ func (m *DynamicScaleUpManager) processDynamicScaleUpReconcileEvent() error {
 		for _, region := range provider.Regions {
 			for supportedInstanceTypeName := range region.SupportedInstanceTypes {
 				supportedInstanceTypeConfig := region.SupportedInstanceTypes[supportedInstanceTypeName]
-				shouldScaleUp, err := m.shouldScaleUpNewClusterForInstanceType(provider.Name, region.Name, supportedInstanceTypeName, &supportedInstanceTypeConfig, kafkaStreamingUnitCountPerClusterList)
+				var dynamicScaleUpProcessor dynamicScaleUpProcessor = &standardDynamicScaleUpProcessor{
+					locator: supportedInstanceTypeLocator{
+						provider:         provider.Name,
+						region:           region.Name,
+						instanceTypeName: supportedInstanceTypeName,
+					},
+					instanceTypeConfig:                    &supportedInstanceTypeConfig,
+					kafkaStreamingUnitCountPerClusterList: kafkaStreamingUnitCountPerClusterList,
+					supportedKafkaInstanceTypesConfig:     &m.KafkaConfig.SupportedInstanceTypes.Configuration,
+					dryRun:                                false,
+				}
+
+				shouldScaleUp, err := dynamicScaleUpProcessor.ShouldScaleUp()
 				if err != nil {
 					return err
 				}
 				if shouldScaleUp {
-					err := m.registerNewClusterForInstanceType(provider.Name, region.Name, supportedInstanceTypeName)
+					err := dynamicScaleUpProcessor.ScaleUp()
 					if err != nil {
 						return err
 					}
```

Comment:
Should we accumalate the list of errors and then send out an `ErrorList` contained the error for each region?
The advantage is that we won't be blocking up scale up evaluation for others regions if we encounter an error in the first regions

Reply:
Thanks. 

We could go with the current way of doing things i.e return a list of `error` i.e `[]error` and we could have a followup on doing it properly with the `ErrorList` type

Reply:
Even with the `ErrorList` I think it needs some more thinking because the issue is the same about the callers

Reply:
I will go with the []error approach for now and think about it

Reply:
here is my thinking thoughts:
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5c2eb2ca43caef5d1e19711bcd8388e58bdc217e/internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go#L92 already returns a list of error.
- we could change the https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5c2eb2ca43caef5d1e19711bcd8388e58bdc217e/internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go#L143 to return the list of `error`, the same for the methods called by this.

We do not need to check for the error type here and once the `.Reconcile()` is called, if errors occurred then they are logged automatically

Later on, on a possibly bing bang refactor, all the reconcilers will be modified to start using `ErrorList` or the CompoundError approach.

Reply:
I wouldn't like to return an slice of errors in the `processDynamicScaleUpReconcileEvent` method.
I will make use of the `ErrorList` approach.

Reply:
I updated the PR.
Now processDynamicScaleUpReconcileEvent does not directly return when one of the region's provider returns an error. Instead, the error is appended to a list of errors and when all of the region's providers have been processed it is returned.
The signature of the method returns an `error` intentionally (as opposed to a []error or an ErrorList).
The ErrorList type has been updated to allow adding errors to it.

Reply:
Thank you Miguel